main.c

```diff
@@ -35,17 +36,58 @@
 #include <libelf.h>
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 #include <unistd.h>
+#include <uk/errptr.h>
 #include <uk/essentials.h>
 #include <uk/plat/memory.h>
 #if CONFIG_LIBPOSIX_PROCESS
 #include <uk/process.h>
 #endif /* CONFIG_LIBPOSIX_PROCESS */
 #include <uk/thread.h>
 #include <uk/sched.h>
+#if CONFIG_LIBUKSWRAND
+#include <uk/swrand.h>
+#endif /* CONFIG_LIBUKSWRAND */
 
 #include "elf_prog.h"
 
+#if CONFIG_LIBPOSIX_ENVIRON
+extern char **environ;
+#else /* !CONFIG_LIBPOSIX_ENVIRON */
+#define environ NULL
+#endif /* !CONFIG_LIBPOSIX_ENVIRON */
+
+#ifndef PAGES2BYTES
+#define PAGES2BYTES(x) ((x) << __PAGE_SHIFT)
+#endif
+
+/*
+ * Inline variant of `basename`
+ * NOTE: This version modifies the input string by overwriting
+ *       trailing slashes.
+ */
+static inline char *_basename(char *path)
```

Comment:
As far as I remember, we preferred "no underscore" but this was also not totally strict. Anyways, I changed it to `basename_internal` because `basename` is in conflict with a GNU version in `string.h` and with a POSIX version in `libgen.h`. Both functions aren't available in `nolibc` and the behavior is also slightly different among them.
([man basename(3)](https://man7.org/linux/man-pages/man3/basename.3.html)).